### DISCUSSION TITLE
/var/lib/pacman not found

### DISCUSSION THREAD

**Discussion by imxade** [termux/proot#291] [Q&A] [answered]
### what I did ?
https://codeberg.org/zz/strap/wiki#on-termux

### following observations are inside of prooted env.

```
ls -ld /var/lib/pacman
drwxr-xr-x 4 root root 4096 Nov 13 01:09 /var/lib/pacman
```

```
pacman -Sy
error: failed to initialize alpm library:
(root: /, dbpath: /var/lib/pacman/)
could not find or read directory
```

```
pacman-db-upgrade
==> ERROR: You must have correct permissions to upgrade the database.
```

**Comment by michalbednarski** [comment]
Looks like this issue is combination of [`realpath("/var/lib/pacman/")` checking if something is directory with faccessat2()](https://github.com/bminor/glibc/blob/f11a92993c45a91c3237733486ce2b8735af2ba0/stdlib/canonicalize.c#L100), [`qemu-user` redirecting `faccessat2()` syscall to faccessat() in libc](https://github.com/qemu/qemu/blob/34a5cb6d8434303c170230644b2a7c1d5781d197/linux-user/syscall.c#L9510) and [Bionic rejecting `faccessat2()` with flags](https://cs.android.com/android/platform/superproject/main/+/main:bionic/libc/bionic/faccessat.cpp;l=45-57;drc=79dbdc3c22d924540c86ed4ea5a7a6945392760b)

I guess we'll (Termux / @xtkoba / me?) have to patch our `qemu` to not pass flags value to `faccessat()`

**Reply by imxade** [reply]
If that's qemu problem should I raise the issue somewhere else too ? [couldn't find the termux-qemu repo]

Also, if that's not an undesired behavior, what changes are to be made on my side ?

**Reply by michalbednarski** [reply, marked as answer]
I've added patch to Termux qemu now (as https://github.com/termux/termux-packages/commit/ef193c0b9c4306fd257e5c6db415f96831cd5972, will appear as version `1:8.0.2-1` once mirrors sync)
